### ISSUE TITLE
train_segmentation_landcover: CPU OOM after epoch 1 on large datasets (val loop accumulates full tensor)

### ISSUE THREAD

**Issue by jayakumarpujar** [opengeos/geoai#717]
## Bug

`train_segmentation_landcover` (via `_train_with_custom_iou`) crashes with SIGKILL at end of epoch 1 on large datasets. Occurs on memory-limited HPC nodes (SLURM, V100 32 GB GPU).

## Root cause

The validation loop accumulates the **entire validation set** as `int64` CPU tensors each epoch:

```python
all_preds = []
all_targets = []
with torch.no_grad():
    for images, targets in val_loader:
        ...
        preds = torch.argmax(outputs, dim=1).cpu()
        all_preds.append(preds)
        all_targets.append(targets.cpu())
all_preds = torch.cat(all_preds, dim=0)
all_targets = torch.cat(all_targets, dim=0)
```

**Memory math** (256×256 tiles, int64 = 8 bytes):
- Per tile: `256 × 256 × 8 × 2` ≈ 1 MB
- 30k–80k val tiles → **30–80 GB CPU RAM** at epoch 1 boundary
- Plus `pin_memory=True` locked RAM + worker fork CoW → exceeds SLURM `--mem` → SIGKILL

## Reproducer

Any training run where val set exceeds ~10k tiles:
```python
train_segmentation_landcover(
    images_dir=...,
    labels_dir=...,  # e.g. 50k tiles total, val_split=0.2 → 10k val tiles
    ...
)
# Killed at end of epoch 1
```

## Fix

Stream a **confusion matrix** (O(num_classes²) memory) for `validation_iou_mode="mean"` instead of collecting full tensors. Other modes retain existing behavior.

PR: https://github.com/jayakumarpujar/geoai-1/pull/new/fix/landcover-val-oom

## Environment

- geoai 0.32.0
- PyTorch 2.x, CUDA 12.x
- SLURM HPC, V100 32 GB GPU, `--mem=64G`
